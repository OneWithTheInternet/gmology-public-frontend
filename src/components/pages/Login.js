import LoginForm from "../organisms/LoginForm";
import { Link } from "react-router-dom";

function Login() {
  return (
    <div className="sectionsContainer">
      <section className="login">
        <h1>Log in</h1>
        <div>
          Create a <Link to="/signup">new account here</Link>
        </div>
        <LoginForm />
      </section>
    </div>
  );
}

export default Login;
