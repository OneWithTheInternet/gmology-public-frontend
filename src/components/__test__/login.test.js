import { render, screen, cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Login from "../pages/Login";

// cleans rendered elements from memory after each test
afterEach(() => {
  cleanup();
});

//Snapshot test for this component
test("should match snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render login page", () => {
  render(
    <MemoryRouter>
      <Login />
    </MemoryRouter>
  );
  const loginform = screen.getByTestId("loginForm");
  expect(loginform).toBeInTheDocument();
});

describe("should login with mock user", () => {
  test("should render email and password form fields", () => {
    render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );
    const emailField = screen.getByTestId("emailField");
    const passwordField = screen.getByTestId("passwordField");
    expect(emailField && passwordField).toBeInTheDocument();
  });
});
