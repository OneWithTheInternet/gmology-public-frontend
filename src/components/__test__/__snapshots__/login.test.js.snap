// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="sectionsContainer"
>
  <section
    className="login"
  >
    <h1>
      Log in
    </h1>
    <div>
      Create a 
      <a
        href="/signup"
        onClick={[Function]}
      >
        new account here
      </a>
    </div>
    <form
      className="login__form"
      data-testid="loginForm"
      onSubmit={[Function]}
    >
      <label>
        Email
        <input
          autoComplete="email"
          data-testid="emailField"
          onChange={[Function]}
          placeholder="example@example.com"
          required={true}
          type="email"
          value=""
        />
      </label>
      <label>
        Password
        <input
          autoComplete="password"
          data-testid="passwordField"
          maxLength={20}
          minLength={6}
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </label>
      <input
        className="submitButton form__button"
        type="submit"
        value="Submit"
      />
      <hr />
      <div>
        <a
          href="/forgot-password"
          onClick={[Function]}
        >
          <p>
            Forgot your password?
          </p>
        </a>
      </div>
    </form>
  </section>
</div>
`;
